recipes/opensubdiv/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+
+    def config_options(self):
+        if self.settings.os != "Windows":
+            del self.options.with_dx
+        if self.settings.os != "Macos":
+            del self.options.with_metal
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            del self.options.shared
+
+    def requirements(self):
+        if self.options.with_tbb:
+            self.requires("onetbb/2020.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    @property
+    def _osd_gpu_enabled(self):
+        return any(
+            [
+                self.options.with_opengl,
+                self.options.with_opencl,
+                self.options.with_cuda,
+                self.options.get_safe("with_dx"),
+                self.options.get_safe("with_metal"),
+            ]
+        )
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["NO_TBB"] = not self.options.with_tbb
+        cmake.definitions["NO_OPENGL"] = not self.options.with_opengl
+        cmake.definitions["BUILD_SHARED_LIBS"] = self.options.get_safe("shared")
+        cmake.definitions["NO_OMP"] = not self.options.with_omp
+        cmake.definitions["NO_CUDA"] = not self.options.with_cuda
+        cmake.definitions["NO_DX"] = not self.options.get_safe("with_dx")
+        cmake.definitions["NO_METAL"] = not self.options.get_safe("with_metal")
+        cmake.definitions["NO_CLEW"] = not self.options.with_clew
+        cmake.definitions["NO_OPENCL"] = not self.options.with_opencl
+        cmake.definitions["NO_PTEX"] = True  # Note: PTEX is for examples only, but we skip them..
+        cmake.definitions["NO_DOC"] = True
+        cmake.definitions["NO_EXAMPLES"] = True
+        cmake.definitions["NO_TUTORIALS"] = True
+        cmake.definitions["NO_REGRESSION"] = True
+        cmake.definitions["NO_TESTS"] = True
+        cmake.definitions["NO_GLTESTS"] = True
+
+        cmake.configure(build_folder=self._build_subfolder)
+
+        return cmake
+
+    def build(self):
+        if not self._osd_gpu_enabled and self.settings.os == "Macos":
+            path = os.path.join(self._source_subfolder, "opensubdiv", "CMakeLists.txt")
+            tools.replace_in_file(path, "$<TARGET_OBJECTS:osd_gpu_obj>", "")
+            tools.replace_in_file(path, "${CMAKE_SOURCE_DIR}/opensubdiv", "${CMAKE_SOURCE_DIR}/source_subfolder/opensubdiv")
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "OpenSubdiv")
+        self.cpp_info.set_property("cmake_target_name", "OpenSubdiv::opensubdiv")
```

Comment:
Do you mean why it's `OpenSubdiv::opensubdiv` instead of `OpenSubdiv::OpenSubdiv`? There is no reason for this, I think I just took the OpenVDB recipe as a reference where its `OpenVDB::openvdb`. So changing this to `OpenSubdiv::OpenSubdiv` should be fine

Reply:
So our goal in CCI is to respect the upstream declarations as closely as possible.

When I looked into opensubdiv I did not find the correct CMake code exporting targets. https://github.com/PixarAnimationStudios/OpenSubdiv/search?l=CMake&q=install+export

It would be ideal to remove these two lines of code so consumers are not present with misleading information

```suggestion
```